src/launcher/java/org/truffleruby/launcher/options/DefaultExecutionAction.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2017 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 1.0
+ * GNU General Public License version 2
+ * GNU Lesser General Public License version 2.1
+ */
+package org.truffleruby.launcher.options;
+
+public enum DefaultExecutionAction {
+
+    NONE {
+        @Override
+        public void applyTo(CommandLineOptions config) {
+            config.setOption(OptionsCatalog.EXECUTION_ACTION, ExecutionAction.NONE);
+        }
+    },
+
+    STDIN {
+        @Override
+        public void applyTo(CommandLineOptions config) {
+            config.setOption(OptionsCatalog.EXECUTION_ACTION, ExecutionAction.STDIN);
+        }
+    },
+
+    IRB {
+        @Override
+        public void applyTo(CommandLineOptions config) {
+            if (System.console() != null) {
+                config.setOption(OptionsCatalog.EXECUTION_ACTION, ExecutionAction.PATH);
+                config.setOption(OptionsCatalog.TO_EXECUTE, "irb");
+
+                System.err.println(
+                        "truffleruby: By default truffleruby drops into IRB instead of reading stdin. " +
+                                "(Use '-' to explicitly read from stdin.)");
```

Comment:
On the other hand since it's printed only when STDIN is tty/interactive it will never be in a way of any scripting (for us or users). It's better to be clear that we are doing something different then MRI, then letting users to figure it out. It might not be too important to have the message. However it's more user friendly, I think it does not make sense to remove it once it's there.